### PR TITLE
Changed Test Clause to CHECK()

### PR DESCRIPTION
**Pull request by ShikharJ**
This is a standing PR for now. It can be updated whenever required (in the future).

### CODE REVIEW

File: symengine/tests/basic/test_arit.cpp [hunk 1]
```diff
@@ -65,85 +65,85 @@ TEST_CASE("Add: arit", "[arit]")
...
     r2 = add(r1, r2);
-    REQUIRE(eq(*r3, *r2));
+    CHECK(eq(*r3, *r2));
 
     r1 = real_double(0.1);
     r2 = Rational::from_mpq(rational_class(1, 2));
     r3 = add(add(add(r1, r2), integer(1)), real_double(0.2));
-    REQUIRE(is_a<RealDouble>(*r3));
-    REQUIRE(std::abs(down_cast<const RealDouble &>(*r3).i - 1.8) < 1e-12);
+    CHECK(is_a<RealDouble>(*r3));
```
Comment (isuruf) [inline symengine/tests/basic/test_arit.cpp]: This is not correct. If this check fails the check below will crash. This has to be a `REQUIRE`

**Comment by isuruf** [comment]
I don't know whether this is really necessary, changing 45 files and creating conflicts with all the other PRs.

**Comment by ShikharJ** [comment]
That's the reason why I mentioned that it is intended as a standing PR, for whenever the need may be (in the future). There had been a number of instances with me, while implementing stuff, where the tests failed but running `ctest --output-on-failure` only returned the name of the test block, and I had to individually check for each failure (which could be anything from a silly typo to an implementation error).

**Comment by rikardn** [comment]
Although this might be a good idea I think due to its many conflicts it will not be worth to resurrect this PR. Can we do some spring cleaning and close?

**Comment by ShikharJ** [comment]
@rikardn I think we should just close. The codebase has changed significantly since, and there isn't a point in resurrecting this, since the potential advantages aren't much over the disadvantages.

**Pull request closed without merging**